src/models/msg.js

```diff
@@ -31,6 +31,7 @@ function Msg(attr) {
 	_.defaults(this, attr, {
 		from: "",
 		id: id++,
+		links: [],
 		previews: [],
```

Comment:
I originally started this with `previews: new Map()`, where each link in the text was added as a key with an `null` value (because `Map` keep key ordering, and they can be easily looped over without some JS heaviness), but it turns out Socket.io transports stuff to the client as JSON so my `Map` was converted to an `Object`.

It doesn't seem like there is a solution for that on the Socket.io side, people deconstruct/reconstruct using arrays of arrays and such, brr. I couldn't be bothered to do some custom/crazy serialization and this worked well enough.